### PR TITLE
Adding read writer

### PR DESCRIPTION
**Pull request by B3rse**
Added ReadWriter to merge and split multiple BAM by regions.
Added read groups handling for TNhaplotyper workflow that can accept multiple BAM files.
Added modified BWA-MEM for aligning Hi-C data

### CODE REVIEW

File: dockerfiles/sentieon/sentieon_ReadWriter.sh [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    rg_lines=$(samtools view -H "$input_bam" | grep "^@RG")
+
+    # Loop through @RG lines and modify ID field
+    # Create --replace_rg arguments
+    while IFS= read -r rg_line; do
+        orig_rg_id=$(echo "$rg_line" | awk -F'\t' '/ID:/ {print $2}' | sed "s/ID://")
+        new_rg_id="${orig_rg_id}-$(generate_random_string)"
+        new_rg_line=$(echo "$rg_line" | cut -f 2- | sed "s/${orig_rg_id}/${new_rg_id}/")
+        formatted_new_rg_line=$(echo -e "$new_rg_line" | sed 's/\t/\\t/g')
+        replace_rg_args+=" --replace_rg ${orig_rg_id}='${formatted_new_rg_line}' "
```
Comment (dmaziec) [inline dockerfiles/sentieon/sentieon_ReadWriter.sh:37]: I think adding a comment for each line would help to understand what we are doing here

File: descriptions/sentieon_merge_TNfilter_normal.cwl [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      position: 1
+    doc: Genome reference in FASTA format with the corresponding index files
+
+  - id: sample_name
+    type: string
+    inputBinding:
+      position: 2
+    doc: Name of the sample
+
+  - id: normal_name
```
Comment (dmaziec) [inline descriptions/sentieon_merge_TNfilter_normal.cwl:38]: this variable name is a bit confusing, maybe `normal_sample_name` would be more clear

File: portal_objects/software.yaml [hunk 3]
```diff
@@ -1,11 +1,41 @@
...
+# This is required to sync with a previously generated object
+uuid: b42e44e5-a829-4687-aeff-65cd040b1528
+
+---
+
+################################################################
+# Sentieon_minimap2, 202308.01
+#   Only used for file naming
+################################################################
+name: Sentieon_minimap2
```
Comment (drio18) [inline portal_objects/software.yaml:35]: Consider adding a UUID, alias, or accession here for consistent reference. Alias may be generated automatically, in which case ignore the comment.

Comment (B3rse) [reply portal_objects/software.yaml:35]: yea, I will link them all by alias. I added the uuid only to the ones you generated manually so to align them on the first deploy

**Pull request merged**